columnflow/plotting/plot_ml_evaluation.py

```diff
@@ -308,3 +308,210 @@ def fmt(v):
+    thresholds = np.linspace(0, 1, n_thresholds)
+    weights = create_sample_weights(sample_weights, events, list(events.keys()))
+    discriminators = list(events.values())[0].fields
+    figs = []
+
+    if evaluation_type not in ["OvO", "OvR"]:
+        raise ValueError(
+            "Illeagal Argument! Evaluation Type can only be choosen as 'OvO' (One vs One)"
+            "or 'OvR' (One vs Rest)",
+        )
+
+    def create_histograms(events: dict, sample_weights: dict, *args, **kwargs) -> dict[str, dict[str, np.ndarray]]:
+        """
+        Helper function to create histograms for the different discriminators and classes.
+        """
+        hists = {}
+        for disc in discriminators:
+            hists[disc] = {}
+            for cls, predictions in events.items():
+                hists[disc][cls] = (sample_weights[cls] *
+                    ak.to_numpy(np.histogram(predictions[disc], bins=thresholds)[0]))
+        return hists
+
+    def binary_roc_data(
+        positiv_hist: np.ndarray,
+        negativ_hist: np.ndarray,
+        *args,
+        **kwargs,
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """
+        Compute binary Receiver operating characteristic (ROC) values.
+        Used as a helper function for the multi-dimensional ROC curve
+        """
+        # calculate the different rates
+        fn = np.cumsum(positiv_hist)
+        tn = np.cumsum(negativ_hist)
+        tp = fn[-1] - fn
+        fp = tn[-1] - tn
+
+        tpr = tp / (tp + fn)
+        fpr = fp / (fp + tn)
+
+        return fpr, tpr
+
+    def roc_curve_data(
+        evaluation_type: str,
+        histograms: dict,
+        *args,
+        **kwargs,
+    ) -> dict[str, dict[str, np.ndarray]]:
+        """
+        Compute Receiver operating characteristic (ROC) values for a multi-dimensional output.
+        """
+        result = {}
+
+        for disc in discriminators:
+            tmp = {}
+
+            if disc in skip_discriminators:
+                continue
+
+            # choose the evaluation type
+            if (evaluation_type == "OvO"):
+                for pos_cls, pos_hist in histograms[disc].items():
+                    for neg_cls, neg_hist in histograms[disc].items():
+
+                        fpr, tpr = binary_roc_data(
+                            positiv_hist=pos_hist,
+                            negativ_hist=neg_hist,
+                            *args,
+                            **kwargs,
+                        )
+                        tmp[f"{pos_cls}_vs_{neg_cls}"] = {"fpr": fpr, "tpr": tpr}
+
+            elif (evaluation_type == "OvR"):
+                for pos_cls, pos_hist in histograms[disc].items():
+                    neg_hist = np.zeros_like(pos_hist)
+                    for neg_cls, neg_pred in histograms[disc].items():
+                        if (pos_cls == neg_cls):
+                            continue
+                        neg_hist += neg_pred
```

Comment:
Just an optional suggestion to de-nest this a bit.

```suggestion
                    neg_hist = sum(
                        (
                            neg_pred
                            for neg_cls, neg_pred in histograms[disc].items()
                            if neg_cls != pos_cls
                        )
                        np.zeros_like(pos_hist),
                    )
```